File: 80.java
// LeetCode 80 Remove Duplicates from Sorted Array II
// Reference: https://www.youtube.com/watch?v=OZaADxYTfD4

class Solution {
    public int removeDuplicates(int[] nums) {
        // TIme O(N)
        // Spce O(1)
        int in = 2;

        if (nums.length < 3) return nums.length;

        for (int i = 2; i < nums.length; i++){
            if (nums[i] != nums[in-2]){
                nums[in] = nums[i];
                in ++;
            }
        }
        return in;
    }
}

File: 80.js
// LeetCode 80 Remove Duplicates from Sorted Array II
// Reference: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/?envType=study-plan-v2&envId=top-interview-150

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let j = 2;

    for (let i = 2; i < nums.length; i++){
        if (nums[i] !== nums[j-2]){
            nums[j++] = nums[i];
        }
    }

    return j;
    
};
        